gh-pages/content/user-guides/lib-user/language-specific/go.md

```diff
@@ -0,0 +1,165 @@
+
+```go
+package main
+
+import (
+  "fmt"
+
+  "example/jsiimodule"
+)
+
+type greeter struct {
+  _ byte // padding
+}
+
+// IMPORTANT - this function has a pointer receiver!
+func (g *greeter) Greet(greetee string) {
+  fmt.Printf("Hello, %s!\n", greetee)
+}
+
+func main() {
+  g := &greeter{}
+
+  // Simply pass the instance though, it "just works".
+  mg := jsiimodule.NewMajestyGreeter(g)
+
+  mg.Announce("Elizabeth II")
+}
+```
+
+## Extending and overriding *classes*
+
+!!! important
+    Leveraging extension and override goes against the design principles of the
+    **go** programming language. We advise you avoid using this mechanism unless
+    you have determined that there is no way to achieve the desired result with
+    composition.
+
+    In particular, if the only element you need to override on a *class* is it's
+    *constructor*, you should simply *decorate* this constructor instead of
+    using the extension and overrides mechanism.
+
+*Classes* that are open for *extension* (including *abstract base classes*) have
+a special *overriding* constructor that can be used when building sub-classes.
+This *override* constructor is expected to be called from within the child
+*class* constructor that you are writing. This constructor is named using the
+following convention: `New<ClassName>_Override`, and receives the *overriding*
+struct instance as the first parameter.
+
+The **go** `struct` that *extends* the base *jsii class* must anonymously embed
+the *jsii class*' **go** interface, while specifying the `overrides:"..."` field
+tag. The value of the tag is a comma-separated list of **go** methods that are
+locally overridden (note: in the case of *property accessors*, the *getter* name
+is to be used, even if only the *setter* is overridden).
+
+Assuming the following abstract base class:
+
+```go
+package jsiimodule
+
+type AbstractBaseClass interface {
+  // Those members have implementations provided, you *may* override them
+  ConcreteMethod() bool
+  ConcreteProperty() string
+  SetConcreteProperty(v string)
+
+
+  // Those members do not have implementations, you *must* implement them
+  AbstractMethod() string
+  AbstractReadonlyProperty() float64
+}
+
+func NewAbstractBaseClass_Override(inst AbstractBaseClass, arg0 string, arg1 float64) {
+  // Omitted for brevity
+}
+```
+
+You can implement that abstract base class in go in the following way:
+
+```go
+package main
+
+import (
+  "fmt"
+  "strings"
+
+  "example/jsiimodule"
+)
+
+type childClass struct {
+  // We implement AbstractMethod, AbstractReadonlyProperty, and override the setter for ConcreteProperty
+  jsiimodule.AbstractBaseClass `overrides:"AbstractMethod,AbstractReadonlyProperty,ConcreteProperty"`
+
+  // Our own storage
+  stringValue string
+}
+
+// Provide your own constructor, which delegates to the base class' overriding
+// constructor.
+func NewChildClass(stringValue string, arg0 string, arg1 float64) jsiimodule.AbstractBaseClass {
```

Comment:
The `arg0`.. names are a bit confusing for the example and may be mistaken as the idiom. I'd use a more concrete set of names.